### PR TITLE
chore: Add GCP to cloud provider

### PR DESCRIPTION
**Pull request by HemaHG**
Signed-off-by: hemahg <hhg@redhat.com>

**What this PR does / why we need it**:

> Add GCP to cloud provider. 

**Which issue(s) this PR fixes** 

fixes https://issues.redhat.com/browse/MGDSTRM-9398

**Link to story**
https://619e1921b3e38b003afeb2be-zyvanquxtu.chromatic.com/?path=/story/kafka-createkafkainstance--create-kafka-instance
https://619e1921b3e38b003afeb2be-zyvanquxtu.chromatic.com/?path=/story/kafka-createkafkainstance-components-fieldcloudprovider--default

### CODE REVIEW

File: src/Kafka/types.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ export type DateIsoString = string;
  * Typescript errors to find where extra information need to be added (images,
  * translations).
  */
-export type CloudProvider = "aws" | "azure";
+export type CloudProvider = "aws" | "gcp" | "azure";
```
Comment (HemaHG) [inline src/Kafka/types.ts:11]: @riccardo-forina adding GCP to the cloud provider type will also affect the MarketPlace billing, MarketPlace type includes `cloudProvider` type, so should we also make changes to the billing MarketPlace billing?

File: src/Kafka/CreateKafkaInstance/components/BillingTiles.tsx [hunk 2]
```diff
@@ -194,5 +196,6 @@ export const BillingTiles: VoidFunctionComponent<BillingTilesProps> = ({
 const tiles: { [id in MarketPlace]: JSX.Element } = {
   aws: <AwsIcon size="md" color="black" />,
   azure: <MicrosoftIcon size="md" color="black" />,
+  gcp: <img src={LogoGCP} alt={""} style={{ height: "50px" }} />,
   rhm: <RedhatIcon size="md" color="black" />,
```
Comment (riccardo-forina) [inline src/Kafka/CreateKafkaInstance/components/BillingTiles.tsx:200]: We should be using the new images here as well for the other providers

File: src/Kafka/CreateKafkaInstance/components/CloudProviderTiles.tsx [hunk 3]
```diff
@@ -86,6 +86,7 @@ export const CloudProvidersTiles: VoidFunctionComponent<
 };
 
 const tiles: { [id in CloudProvider]: JSX.Element } = {
-  aws: <AwsIcon size="lg" color="black" />,
-  azure: <MicrosoftIcon size="lg" color="black" />,
+  aws: <img src={LogoAWS} alt={""} style={{ height: "60px" }} />,
+  gcp: <img src={LogoGCP} alt={""} style={{ height: "60px" }} />,
+  azure: <img src={LogoAzure} alt={""} style={{ height: "60px" }} />,
```
Comment (riccardo-forina) [inline src/Kafka/CreateKafkaInstance/components/CloudProviderTiles.tsx:91]: I feel like these images would make a great candidate to become components in the shared folder, so to use them like `<AWSLogo />` or something like that

File: src/Kafka/types.ts (hunk 1 above)
Comment (riccardo-forina) [reply src/Kafka/types.ts:11]: yes it's ok for GCP to be in the marketplace as well, it will not be displayed unless the API says so

**Review by riccardo-forina — COMMENTED** [review]
Thank you, it looks good. I left a few comments

File: src/Kafka/CreateKafkaInstance/Stories/storiesHelpers.tsx [hunk 4]
```diff
@@ -53,7 +53,13 @@ const AZURE: CloudProviderInfo = {
   ],
 };
 
-export const PROVIDERS: CloudProviderInfo[] = [AWS, AZURE];
+const GCP: CloudProviderInfo = {
+  id: "gcp",
+  displayName: "Google cloud platform",
+  regions: [],
```
Comment (riccardo-forina) [inline src/Kafka/CreateKafkaInstance/Stories/storiesHelpers.tsx:59]: Could you please add some sample region? So to make the initial example work when selecting GCP.

File: src/Kafka/CreateKafkaInstance/components/BillingTiles.tsx [hunk 5]
```diff
@@ -192,7 +194,8 @@ export const BillingTiles: VoidFunctionComponent<BillingTilesProps> = ({
 };
 
 const tiles: { [id in MarketPlace]: JSX.Element } = {
-  aws: <AwsIcon size="md" color="black" />,
-  azure: <MicrosoftIcon size="md" color="black" />,
+  aws: <AWSLogo />,
+  gcp: <GCPLogo />,
+  azure: <AzureLogo />,
   rhm: <RedhatIcon size="md" color="black" />,
```
Comment (HemaHG) [inline src/Kafka/CreateKafkaInstance/components/BillingTiles.tsx:200]: @jgiardino @riccardo-forina Do you guess know from where I can get image for Redhat logo?

**Review by anukritijha — APPROVED** [review]

**Pull request merged**